compiler/rustc_codegen_cranelift/src/intrinsics/simd.rs

```diff
@@ -96,10 +98,11 @@ pub(super) fn codegen_simd_intrinsic_call<'tcx>(
 
                 (0..ret_lane_count).map(|i| {
                     let i = usize::try_from(i).unwrap();
-                    let idx = rustc_middle::mir::interpret::read_target_uint(
-                        fx.tcx.data_layout.endian,
-                        &idx_bytes[4*i.. 4*i + 4],
-                    ).expect("read_target_uint");
+                    let idx = match (fx.tcx.data_layout.endian, &idx_bytes[4*i.. 4*i + 4]) {
+                        (Endian::Little, &[a, b, c, d]) => u32::from_le_bytes([a, b, c, d]),
+                        (Endian::Big, &[a, b, c, d]) => u32::from_be_bytes([a, b, c, d]),
+                        (_, _) => unreachable!(),
+                    };
```

Comment:
This has become significantly more complex when the complexity was hidden behind a function before. Why was this change made?